sympy/concrete/products.py

```diff
@@ -399,6 +399,13 @@ def _eval_product_direct(self, term, limits):
         (k, a, n) = limits
         return Mul(*[term.subs(k, a + i) for i in range(n - a + 1)])
 
+    def _eval_derivative(self, x):
+        from sympy import Sum
```

Comment:
This should be at the top of the file. We generally use local imports to avoid cyclic imports but I don't think that will be a problem here.

Reply:
Sure, will do.
Thanks.

Reply:
It has already been defined as `from sympy.concrete.summations import Sum` inside many methods, thus introducing it in above leads to circular imports, Should I also define it as the same inside `_eval_derivative`?

Reply:
All right. Probably best to keep it how it is for now: `from sympy.concrete.summations import Sum`.